src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpOverallOutcomeController.cs

```diff
@@ -0,0 +1,102 @@
+        }
+        
+        [HttpGet]
+        [Authorize(Policy = "AccessApplication")]
+        public async Task<IActionResult> ProcessApplicationStatus(Guid applicationId)
+        {
+            var application = await _applicationApiClient.GetApplication(applicationId);
+            var model = _service.BuildApplicationSummaryViewModel(application, User.GetEmail());
+            var applicationStatus = application.ApplicationStatus;
+
+            switch (applicationStatus)
+            {
+                case ApplicationStatus.New:
+                case ApplicationStatus.InProgress:
+                    return RedirectToAction("TaskList", "RoatpApplication", new {applicationId});
+                case ApplicationStatus.Approved:
+                {
+                    var oversightReview = await _apiClient.GetOversightReview(applicationId);
+                    if (oversightReview?.Status == OversightReviewStatus.SuccessfulAlreadyActive)
+                        return View("~/Views/Roatp/ApplicationApprovedAlreadyActive.cshtml", model);
+
+                    return View("~/Views/Roatp/ApplicationApproved.cshtml", model);
+                }
+                case ApplicationStatus.Rejected: //this logic will need to change with the coming status update story
+                    if (application.GatewayReviewStatus == GatewayReviewStatus.Fail)
+                    {
+                        return View("~/Views/Roatp/ApplicationUnsuccessful.cshtml", model);
+                    }
+
+                    var unsuccessfulModel = await _service.ApplicationUnsuccessful(application, User.GetEmail());
+                    return View("~/Views/Roatp/ApplicationUnsuccessfulPostGateway.cshtml", unsuccessfulModel);
+
+                case ApplicationStatus.FeedbackAdded:
+                    return View("~/Views/Roatp/FeedbackAdded.cshtml", model);
+                case ApplicationStatus.Withdrawn:
+                    return View("~/Views/Roatp/ApplicationWithdrawn.cshtml", model);
+                case ApplicationStatus.Removed:
+                    return View("~/Views/Roatp/ApplicationWithdrawnESFA.cshtml", model);
+                case ApplicationStatus.GatewayAssessed:
+                    if (application.GatewayReviewStatus == GatewayReviewStatus.Reject)
+                        return View("~/Views/Roatp/ApplicationRejected.cshtml", model);
+                    return View("~/Views/Roatp/ApplicationSubmitted.cshtml", model);
+                case ApplicationStatus.Submitted:
+                case ApplicationStatus.Resubmitted:
+                    return View("~/Views/Roatp/ApplicationSubmitted.cshtml", model);
+                default:
+                    return RedirectToAction("TaskList", "RoatpApplication", new {applicationId});
+            }
+        }
+
+        public async Task<IActionResult> DownloadFile(Guid applicationId, int sequenceNo, int sectionNo,
+            string pageId, string questionId, string filename)
+        {
```

Comment:
Is this code a repeat of what's in `ApplicationController`? Other than requiring a slightly different state (submitted rather than in progress). Would be better to tweak the policy than copy/paste the code imo.

Reply:
Looks to me like it is, but perhaps in a different context? If that is the case then it might be better keeping the separation of concerns.

Reply:
Not sure where the separation of concerns comes into it. The user only needs one endpoint with which to download files, right?